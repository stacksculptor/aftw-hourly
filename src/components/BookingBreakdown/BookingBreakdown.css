@import '../../marketplace.css';

.root {
  display: flex;
  flex-direction: column;
}

.nowrap {
  white-space: nowrap;
}

.lineItem {
  @apply --marketplaceH4FontStyles;

  /* Since the line items have different font sizes in the label and the
   value columns, we have to balance the line to align to the
   baseline. */
  margin: -1px 0 0 0;

  @media (--viewportMedium) {
    margin: -1px 0 8px 0;
  }

  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.itemLabel {
  margin-top: 1px; /* align with baseline */
}

.itemValue {
  @apply --marketplaceDefaultFontStyles;
  font-weight: 600;
  margin: 0;

  @media (--viewportMedium) {
    font-weight: 700;
    margin-top: 1px; /* align with baseline */
    margin-bottom: 0;
  }
}

.totalDivider {
  /* dimensions */
  width: 100%;
  height: 1px;
  margin: 13px 0 4px 0;

  border: none;
  background-color: var(--matterColorNegative);

  @media (--viewportMedium) {
    margin: 2px 0 8px 0;
  }
}

.totalLabel {
  font-weight: var(--fontWeightSemiBold);

  /* Move so that baseline aligns with the total price */
  padding-top: 7px;
}

.totalPrice {
  @apply --marketplaceH2FontStyles;
  margin: 0;

  @media (--viewportMedium) {
    margin: 0;
  }
}
