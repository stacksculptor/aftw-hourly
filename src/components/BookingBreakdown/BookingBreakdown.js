/**
 * This component will show the booking info and calculated total price.
 * I.e. dates and other details related to payment decision in receipt format.
 */
import React, { PropTypes } from 'react';
import { FormattedMessage, FormattedHTMLMessage, intlShape, injectIntl } from 'react-intl';
import classNames from 'classnames';
import { types } from '../../util/sdkLoader';
import { formatMoney } from '../../util/currency';
import * as propTypes from '../../util/propTypes';
import { error as logError } from '../../util/log';

import css from './BookingBreakdown.css';

const { Money } = types;

// Validate the assumption that the commission exists and the amount
// is zero or negative.
const isValidCommission = commissionLineItem => {
  return commissionLineItem &&
    commissionLineItem.lineTotal instanceof Money &&
    commissionLineItem.lineTotal.amount <= 0;
};

export const BookingBreakdownComponent = props => {
  const {
    rootClassName,
    className,
    userRole,
    transaction,
    booking,
    intl,
  } = props;

  const isProvider = userRole === 'provider';
  const classes = classNames(rootClassName || css.root, className);

  const dateFormatOptions = {
    weekday: 'short',
    month: 'short',
    day: 'numeric',
  };
  const bookingPeriod = (
    <FormattedMessage
      id="BookingBreakdown.bookingPeriod"
      values={{
        bookingStart: (
          <span className={css.nowrap}>
            {intl.formatDate(booking.attributes.start, dateFormatOptions)}
          </span>
        ),
        bookingEnd: (
          <span className={css.nowrap}>
            {intl.formatDate(booking.attributes.end, dateFormatOptions)}
          </span>
        ),
      }}
    />
  );

  const nightPurchase = transaction.attributes.lineItems.find(
    item => item.code === 'line-item/night'
  );
  const providerCommissionLineItem = transaction.attributes.lineItems.find(
    item => item.code === 'line-item/provider-commission'
  );

  const nightCount = nightPurchase.quantity.toFixed();
  const nightCountMessage = (
    <FormattedHTMLMessage id="BookingBreakdown.nightCount" values={{ count: nightCount }} />
  );

  const formattedUnitPrice = formatMoney(intl, nightPurchase.unitPrice);

  // If commission is passed it will be shown as a fee already reduces from the total price
  let subTotalInfo = null;
  let commissionInfo = null;

  if (isProvider) {
    if (!isValidCommission(providerCommissionLineItem)) {
      const error = new Error(
        'Commission should be present and the value should be zero or negative'
      );
      logError(error, {
        providerCommissionLineItem: providerCommissionLineItem,
        transaction: transaction,
      });
      throw error;
    }

    // The total sum that the customer pays is the payinTotal. We use
    // this amount as the subtotal information for the provider
    // breakdown.
    //
    // NOTE: this might break if the API hides the payinTotal from the provider
    const formattedSubTotal = formatMoney(intl, transaction.attributes.payinTotal);
    subTotalInfo = (
      <div className={css.lineItem}>
        <span className={css.itemLabel}>
          <FormattedMessage id="BookingBreakdown.subTotal" />
        </span>
        <span className={css.itemValue}>{formattedSubTotal}</span>
      </div>
    );

    const commission = providerCommissionLineItem.lineTotal;
    const formattedCommission = commission ? formatMoney(intl, commission) : null;

    commissionInfo = (
      <div className={css.lineItem}>
        <span className={css.itemLabel}>
          <FormattedMessage id="BookingBreakdown.commission" />
        </span>
        <span className={css.itemValue}>{formattedCommission}</span>
      </div>
    );
  }

  let providerTotalMessageId = 'BookingBreakdown.providerTotalDefault';
  if (propTypes.txIsDelivered(transaction)) {
    providerTotalMessageId = 'BookingBreakdown.providerTotalDelivered';
  } else if (propTypes.txIsRejectedOrAutorejected(transaction)) {
    providerTotalMessageId = 'BookingBreakdown.providerTotalRejected';
  }

  const totalLabel = isProvider
    ? <FormattedMessage id={providerTotalMessageId} />
    : <FormattedMessage id="BookingBreakdown.total" />;

  const totalPrice = isProvider
    ? transaction.attributes.payoutTotal
    : transaction.attributes.payinTotal;
  const formattedTotalPrice = formatMoney(intl, totalPrice);

  return (
    <div className={classes}>
      <div className={css.lineItem}>
        <span className={css.itemLabel}>
          <FormattedMessage id="BookingBreakdown.pricePerNight" />
        </span>
        <span className={css.itemValue}>{formattedUnitPrice}</span>
      </div>
      <div className={css.lineItem}>
        <span className={css.itemLabel}>
          {bookingPeriod}
        </span>
        <span className={css.itemValue}>{nightCountMessage}</span>
      </div>
      {subTotalInfo}
      {commissionInfo}
      <hr className={css.totalDivider} />
      <div className={css.lineItem}>
        <div className={css.totalLabel}>
          {totalLabel}
        </div>
        <div className={css.totalPrice}>
          {formattedTotalPrice}
        </div>
      </div>
    </div>
  );
};

BookingBreakdownComponent.defaultProps = { rootClassName: null, className: null };

const { oneOf, string } = PropTypes;

BookingBreakdownComponent.propTypes = {
  rootClassName: string,
  className: string,

  userRole: oneOf(['customer', 'provider']).isRequired,
  transaction: propTypes.transaction.isRequired,
  booking: propTypes.booking.isRequired,

  // from injectIntl
  intl: intlShape.isRequired,
};

const BookingBreakdown = injectIntl(BookingBreakdownComponent);

BookingBreakdown.displayName = 'BookingBreakdown';

export default BookingBreakdown;
