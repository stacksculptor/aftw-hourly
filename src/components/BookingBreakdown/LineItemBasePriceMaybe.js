import React from 'react';
import { FormattedMessage, intlShape } from '../../util/reactIntl';
import { formatMoney } from '../../util/currency';
import { LINE_ITEM_NIGHT, LINE_ITEM_DAY, propTypes } from '../../util/types';

import css from './BookingBreakdown.css';

const LineItemBasePriceMaybe = props => {
  const { transaction, unitType, intl } = props;
  const isNightly = unitType === LINE_ITEM_NIGHT;
  const isDaily = unitType === LINE_ITEM_DAY;
  const translationKey = isNightly
    ? 'BookingBreakdown.baseUnitNight'
    : isDaily
    ? 'BookingBreakdown.baseUnitDay'
    : 'BookingBreakdown.baseUnitQuantity';

  const unitPurchase = transaction.attributes.lineItems.find(
    item => item.code === unitType && !item.reversal
  );

  const quantity = unitPurchase ? unitPurchase.quantity.toString() : null;
  const unitPrice = unitPurchase ? formatMoney(intl, unitPurchase.unitPrice) : null;
  const total = unitPurchase ? formatMoney(intl, unitPurchase.lineTotal) : null;

  return quantity && total ? (
    <div className={css.lineItem}>
      <span className={css.itemLabel}>
        <FormattedMessage id={translationKey} values={{ unitPrice, quantity }} />
      </span>
      <span className={css.itemValue}>{total}</span>
    </div>
  ) : null;
};

LineItemBasePriceMaybe.propTypes = {
  transaction: propTypes.transaction.isRequired,
  unitType: propTypes.bookingUnitType.isRequired,
  intl: intlShape.isRequired,
};

export default LineItemBasePriceMaybe;
