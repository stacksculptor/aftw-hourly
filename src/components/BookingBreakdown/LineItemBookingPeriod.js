import React from 'react';
import { FormattedMessage, FormattedHTMLMessage, FormattedDate } from 'react-intl';
import moment from 'moment';
import { LINE_ITEM_NIGHT, propTypes } from '../../util/types';
import { daysBetween } from '../../util/dates';

import css from './BookingBreakdown.css';

const BookingPeriod = props => {
  const { isSingleDay, startDate, endDate } = props;
  const dateFormatOptions = {
    weekday: 'short',
    month: 'short',
    day: 'numeric',
  };

  if (isSingleDay) {
    return <FormattedDate value={startDate} {...dateFormatOptions} />;
  }

  return (
    <FormattedMessage
      id="BookingBreakdown.bookingPeriod"
      values={{
        bookingStart: (
          <span className={css.nowrap}>
            <FormattedDate value={startDate} {...dateFormatOptions} />
          </span>
        ),
        bookingEnd: (
          <span className={css.nowrap}>
            <FormattedDate value={endDate} {...dateFormatOptions} />
          </span>
        ),
      }}
    />
  );
};

const LineItemBookingPeriod = props => {
  const { transaction, booking, unitType } = props;

  const { start: startDate, end: endDateRaw } = booking.attributes;
  const isNightly = unitType === LINE_ITEM_NIGHT;
  const isDaily = unitType === LINE_ITEM_NIGHT;

  const dayCount = daysBetween(startDate, endDateRaw);
  const isSingleDay = !isNightly && dayCount === 1;
  const endDay = isNightly ? endDateRaw : moment(endDateRaw).subtract(1, 'days');

  const unitPurchase = transaction.attributes.lineItems.find(
    item => item.code === unitType && !item.reversal
  );

  const dayCountForLineItemUnits = dayCount + 1;
  const count = isNightly || isDaily ? unitPurchase.quantity.toFixed() : dayCountForLineItemUnits;

  const unitCountMessage = (
    <FormattedHTMLMessage
      id={isNightly ? 'BookingBreakdown.nightCount' : 'BookingBreakdown.dayCount'}
      values={{ count }}
    />
  );

  return (
    <div className={css.lineItem}>
      <span className={css.itemLabel}>
        <BookingPeriod isSingleDay={isSingleDay} startDate={startDate} endDate={endDay} />
      </span>
      <span className={css.itemValue}>{unitCountMessage}</span>
    </div>
  );
};

LineItemBookingPeriod.propTypes = {
  transaction: propTypes.transaction.isRequired,
  booking: propTypes.booking.isRequired,
};

export default LineItemBookingPeriod;
