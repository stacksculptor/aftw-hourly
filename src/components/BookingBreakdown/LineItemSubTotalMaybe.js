import React from 'react';
import { bool } from 'prop-types';
import { FormattedMessage, intlShape } from 'react-intl';
import { formatMoney } from '../../util/currency';
import { propTypes } from '../../util/types';

import css from './BookingBreakdown.css';

const LineItemSubTotalMaybe = props => {
  const { transaction, unitType, isProvider, intl } = props;

  const refund = transaction.attributes.lineItems.find(
    item => item.code === unitType && item.reversal
  );

  // Show unit purchase line total (unit price * quantity) as a subtotal.
  // PLEASE NOTE that this assumes that the transaction doesn't have other
  // line item types than the defined unit type (e.g. week, month, year).
  const showSubTotal = isProvider || refund;
  const unitPurchase = transaction.attributes.lineItems.find(
    item => item.code === unitType && !item.reversal
  );
  const formattedSubTotal = formatMoney(intl, unitPurchase.lineTotal);

  return showSubTotal ? (
    <div className={css.lineItem}>
      <span className={css.itemLabel}>
        <FormattedMessage id="BookingBreakdown.subTotal" />
      </span>
      <span className={css.itemValue}>{formattedSubTotal}</span>
    </div>
  ) : null;
};

LineItemSubTotalMaybe.propTypes = {
  transaction: propTypes.transaction.isRequired,
  isProvider: bool.isRequired,
  intl: intlShape.isRequired,
};

export default LineItemSubTotalMaybe;
