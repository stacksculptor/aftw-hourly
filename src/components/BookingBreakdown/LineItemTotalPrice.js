import React from 'react';
import { bool } from 'prop-types';
import { FormattedMessage, intlShape } from 'react-intl';
import { formatMoney } from '../../util/currency';
import { txIsCanceled, txIsCompleted, txIsDeclinedOrExpired, propTypes } from '../../util/types';

import css from './BookingBreakdown.css';

const LineItemUnitPrice = props => {
  const { transaction, isProvider, intl } = props;

  let providerTotalMessageId = 'BookingBreakdown.providerTotalDefault';
  if (txIsCompleted(transaction)) {
    providerTotalMessageId = 'BookingBreakdown.providerTotalDelivered';
  } else if (txIsDeclinedOrExpired(transaction)) {
    providerTotalMessageId = 'BookingBreakdown.providerTotalDeclined';
  } else if (txIsCanceled(transaction)) {
    providerTotalMessageId = 'BookingBreakdown.providerTotalCanceled';
  }

  const totalLabel = isProvider ? (
    <FormattedMessage id={providerTotalMessageId} />
  ) : (
    <FormattedMessage id="BookingBreakdown.total" />
  );

  const totalPrice = isProvider
    ? transaction.attributes.payoutTotal
    : transaction.attributes.payinTotal;
  const formattedTotalPrice = formatMoney(intl, totalPrice);

  return (
    <div className={css.lineItem}>
      <div className={css.totalLabel}>{totalLabel}</div>
      <div className={css.totalPrice}>{formattedTotalPrice}</div>
    </div>
  );
};

LineItemUnitPrice.propTypes = {
  transaction: propTypes.transaction.isRequired,
  isProvider: bool.isRequired,
  intl: intlShape.isRequired,
};

export default LineItemUnitPrice;
