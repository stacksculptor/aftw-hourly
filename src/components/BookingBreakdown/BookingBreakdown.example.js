import Decimal from 'decimal.js';
import { types } from '../../util/sdkLoader';
import * as propTypes from '../../util/propTypes';
import BookingBreakdown from './BookingBreakdown';

const { UUID, Money } = types;

const exampleBooking = attributes => {
  return {
    id: new UUID('example-booking'),
    type: 'booking',
    attributes,
  };
};

const exampleTransaction = params => {
  const created = new Date(Date.UTC(2017, 1, 1));
  return {
    id: new UUID('example-transaction'),
    type: 'transaction',
    attributes: {
      createdAt: created,
      lastTransitionedAt: created,
      lastTransition: propTypes.TX_TRANSITION_PREAUTHORIZE,

      // payinTotal, payoutTotal, and lineItems required in params
      ...params,
    },
  };
};

export const Checkout = {
  component: BookingBreakdown,
  props: {
    userRole: 'customer',
    transaction: exampleTransaction({
      payinTotal: new Money(9000, 'USD'),
      payoutTotal: new Money(9000, 'USD'),
      lineItems: [
        {
          code: 'line-item/night',
          quantity: new Decimal(2),
          unitPrice: new Money(4500, 'USD'),
          lineTotal: new Money(9000, 'USD'),
        },
      ],
    }),
    booking: exampleBooking({
      start: new Date(Date.UTC(2017, 3, 14)),
      end: new Date(Date.UTC(2017, 3, 16)),
    }),
  },
};

export const CustomerOrder = {
  component: BookingBreakdown,
  props: {
    userRole: 'customer',
    transaction: exampleTransaction({
      payinTotal: new Money(9000, 'USD'),
      payoutTotal: new Money(9000, 'USD'),
      lineItems: [
        {
          code: 'line-item/night',
          quantity: new Decimal(2),
          unitPrice: new Money(4500, 'USD'),
          lineTotal: new Money(9000, 'USD'),
        },
      ],
    }),
    booking: exampleBooking({
      start: new Date(Date.UTC(2017, 3, 14)),
      end: new Date(Date.UTC(2017, 3, 16)),
    }),
  },
};

export const ProviderSale = {
  component: BookingBreakdown,
  props: {
    userRole: 'provider',
    transaction: exampleTransaction({
      payinTotal: new Money(9000, 'USD'),
      payoutTotal: new Money(7000, 'USD'),
      lineItems: [
        {
          code: 'line-item/night',
          quantity: new Decimal(2),
          unitPrice: new Money(4500, 'USD'),
          lineTotal: new Money(9000, 'USD'),
        },
        {
          code: 'line-item/provider-commission',
          unitPrice: new Money(-2000, 'USD'),
          lineTotal: new Money(-2000, 'USD'),
        },
      ],
    }),
    booking: exampleBooking({
      start: new Date(Date.UTC(2017, 3, 14)),
      end: new Date(Date.UTC(2017, 3, 16)),
    }),
  },
};

export const ProviderSaleZeroCommission = {
  component: BookingBreakdown,
  props: {
    userRole: 'provider',
    transaction: exampleTransaction({
      payinTotal: new Money(9000, 'USD'),
      payoutTotal: new Money(9000, 'USD'),
      lineItems: [
        {
          code: 'line-item/night',
          quantity: new Decimal(2),
          unitPrice: new Money(4500, 'USD'),
          lineTotal: new Money(9000, 'USD'),
        },
        {
          code: 'line-item/provider-commission',
          unitPrice: new Money(0, 'USD'),
          lineTotal: new Money(0, 'USD'),
        },
      ],
    }),
    booking: exampleBooking({
      start: new Date(Date.UTC(2017, 3, 14)),
      end: new Date(Date.UTC(2017, 3, 16)),
    }),
  },
};

export const ProviderSaleSingleNight = {
  component: BookingBreakdown,
  props: {
    userRole: 'provider',
    transaction: exampleTransaction({
      payinTotal: new Money(4500, 'USD'),
      payoutTotal: new Money(2500, 'USD'),
      lineItems: [
        {
          code: 'line-item/night',
          quantity: new Decimal(1),
          unitPrice: new Money(4500, 'USD'),
          lineTotal: new Money(4500, 'USD'),
        },
        {
          code: 'line-item/provider-commission',
          unitPrice: new Money(-2000, 'USD'),
          lineTotal: new Money(-2000, 'USD'),
        },
      ],
    }),
    booking: exampleBooking({
      start: new Date(Date.UTC(2017, 3, 14)),
      end: new Date(Date.UTC(2017, 3, 15)),
    }),
  },
};

export const ProviderSaleRejected = {
  component: BookingBreakdown,
  props: {
    userRole: 'provider',
    transaction: exampleTransaction({
      lastTransition: propTypes.TX_TRANSITION_REJECT,
      payinTotal: new Money(4500, 'USD'),
      payoutTotal: new Money(2500, 'USD'),
      lineItems: [
        {
          code: 'line-item/night',
          quantity: new Decimal(1),
          unitPrice: new Money(4500, 'USD'),
          lineTotal: new Money(4500, 'USD'),
        },
        {
          code: 'line-item/provider-commission',
          unitPrice: new Money(-2000, 'USD'),
          lineTotal: new Money(-2000, 'USD'),
        },
      ],
    }),
    booking: exampleBooking({
      start: new Date(Date.UTC(2017, 3, 14)),
      end: new Date(Date.UTC(2017, 3, 15)),
    }),
  },
};

export const ProviderSaleDelivered = {
  component: BookingBreakdown,
  props: {
    userRole: 'provider',
    transaction: exampleTransaction({
      lastTransition: propTypes.TX_TRANSITION_MARK_DELIVERED,
      payinTotal: new Money(4500, 'USD'),
      payoutTotal: new Money(2500, 'USD'),
      lineItems: [
        {
          code: 'line-item/night',
          quantity: new Decimal(1),
          unitPrice: new Money(4500, 'USD'),
          lineTotal: new Money(4500, 'USD'),
        },
        {
          code: 'line-item/provider-commission',
          unitPrice: new Money(-2000, 'USD'),
          lineTotal: new Money(-2000, 'USD'),
        },
      ],
    }),
    booking: exampleBooking({
      start: new Date(Date.UTC(2017, 3, 14)),
      end: new Date(Date.UTC(2017, 3, 15)),
    }),
  },
};
