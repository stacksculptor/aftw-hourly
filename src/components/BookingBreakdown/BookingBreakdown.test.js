import React from 'react';
import Decimal from 'decimal.js';
import { fakeIntl } from '../../util/test-data';
import { renderDeep } from '../../util/test-helpers';
import { types } from '../../util/sdkLoader';
import * as propTypes from '../../util/propTypes';
import { BookingBreakdownComponent } from './BookingBreakdown';

const { UUID, Money } = types;

const exampleBooking = attributes => {
  return {
    id: new UUID('example-booking'),
    type: 'booking',
    attributes,
  };
};

const exampleTransaction = params => {
  const created = new Date(Date.UTC(2017, 1, 1));
  return {
    id: new UUID('example-transaction'),
    type: 'transaction',
    attributes: {
      createdAt: created,
      lastTransitionedAt: created,
      lastTransition: propTypes.TX_TRANSITION_PREAUTHORIZE,
      state: propTypes.TX_STATE_PREAUTHORIZED,

      // payinTotal, payoutTotal, and lineItems required in params
      ...params,
    },
  };
};

describe('BookingBreakdown', () => {
  it('pretransaction data matches snapshot', () => {
    const tree = renderDeep(
      <BookingBreakdownComponent
        userRole="customer"
        transaction={exampleTransaction({
          payinTotal: new Money(2000, 'USD'),
          payoutTotal: new Money(2000, 'USD'),
          lineItems: [
            {
              code: 'line-item/night',
              quantity: new Decimal(2),
              lineTotal: new Money(2000, 'USD'),
              unitPrice: new Money(1000, 'USD'),
            },
          ],
        })}
        booking={exampleBooking({
          start: new Date(Date.UTC(2017, 3, 14)),
          end: new Date(Date.UTC(2017, 3, 16)),
        })}
        intl={fakeIntl}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('customer transaction data matches snapshot', () => {
    const tree = renderDeep(
      <BookingBreakdownComponent
        userRole="customer"
        transaction={exampleTransaction({
          payinTotal: new Money(2000, 'USD'),
          payoutTotal: new Money(2000, 'USD'),
          lineItems: [
            {
              code: 'line-item/night',
              quantity: new Decimal(2),
              lineTotal: new Money(2000, 'USD'),
              unitPrice: new Money(1000, 'USD'),
            },
          ],
        })}
        booking={exampleBooking({
          start: new Date(Date.UTC(2017, 3, 14)),
          end: new Date(Date.UTC(2017, 3, 16)),
        })}
        intl={fakeIntl}
      />
    );
    expect(tree).toMatchSnapshot();
  });

  it('provider transaction data matches snapshot', () => {
    const tree = renderDeep(
      <BookingBreakdownComponent
        userRole="provider"
        transaction={exampleTransaction({
          payinTotal: new Money(2000, 'USD'),
          payoutTotal: new Money(1800, 'USD'),
          lineItems: [
            {
              code: 'line-item/night',
              quantity: new Decimal(2),
              lineTotal: new Money(2000, 'USD'),
              unitPrice: new Money(1000, 'USD'),
            },
            {
              code: 'line-item/provider-commission',
              lineTotal: new Money(-200, 'USD'),
              unitPrice: new Money(-200, 'USD'),
            },
          ],
        })}
        booking={exampleBooking({
          start: new Date(Date.UTC(2017, 3, 14)),
          end: new Date(Date.UTC(2017, 3, 16)),
        })}
        intl={fakeIntl}
      />
    );
    expect(tree).toMatchSnapshot();
  });
});
