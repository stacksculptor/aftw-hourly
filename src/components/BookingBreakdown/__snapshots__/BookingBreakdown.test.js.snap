exports[`BookingBreakdown customer transaction data matches snapshot 1`] = `
<div
  className="">
  <div
    className={undefined}>
    <span
      className={undefined}>
      <span>
        Price per night
      </span>
    </span>
    <span
      className={undefined}>
      10
    </span>
  </div>
  <div
    className={undefined}>
    <span
      className={undefined}>
      <span>
        <span
          className={undefined}>
          2017-04-14
        </span>
         – 
        <span
          className={undefined}>
          2017-04-16
        </span>
      </span>
    </span>
    <span
      className={undefined}>
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&times; 2 nights",
          }
        } />
    </span>
  </div>
  <hr
    className={undefined} />
  <div
    className={undefined}>
    <div
      className={undefined}>
      <span>
        Total price
      </span>
    </div>
    <div
      className={undefined}>
      20
    </div>
  </div>
</div>
`;

exports[`BookingBreakdown pretransaction data matches snapshot 1`] = `
<div
  className="">
  <div
    className={undefined}>
    <span
      className={undefined}>
      <span>
        Price per night
      </span>
    </span>
    <span
      className={undefined}>
      10
    </span>
  </div>
  <div
    className={undefined}>
    <span
      className={undefined}>
      <span>
        <span
          className={undefined}>
          2017-04-14
        </span>
         – 
        <span
          className={undefined}>
          2017-04-16
        </span>
      </span>
    </span>
    <span
      className={undefined}>
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&times; 2 nights",
          }
        } />
    </span>
  </div>
  <hr
    className={undefined} />
  <div
    className={undefined}>
    <div
      className={undefined}>
      <span>
        Total price
      </span>
    </div>
    <div
      className={undefined}>
      20
    </div>
  </div>
</div>
`;

exports[`BookingBreakdown provider transaction data matches snapshot 1`] = `
<div
  className="">
  <div
    className={undefined}>
    <span
      className={undefined}>
      <span>
        Price per night
      </span>
    </span>
    <span
      className={undefined}>
      10
    </span>
  </div>
  <div
    className={undefined}>
    <span
      className={undefined}>
      <span>
        <span
          className={undefined}>
          2017-04-14
        </span>
         – 
        <span
          className={undefined}>
          2017-04-16
        </span>
      </span>
    </span>
    <span
      className={undefined}>
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&times; 2 nights",
          }
        } />
    </span>
  </div>
  <div
    className={undefined}>
    <span
      className={undefined}>
      <span>
        Saunatime fee:
      </span>
    </span>
    <span
      className={undefined}>
      2
    </span>
  </div>
  <hr
    className={undefined} />
  <div
    className={undefined}>
    <div
      className={undefined}>
      <span>
        Total price
      </span>
    </div>
    <div
      className={undefined}>
      18
    </div>
  </div>
</div>
`;
